PR to simulate real-world environment

Rohita83 opened the pull request:
**Description of changes:**

This PR include changes made to simulate real-world environment.  Added Gazebo environment (Model Training) with brick wall and Noise filter in inference at robot_ws.  

**Steps to create Gazebo environment (Model Training) with brick wall**

Goto "aws-robomaker-sample-application-objecttracker/src/object_tracker_simulation/worlds/" Path, set the environment variables and then execute below command to create brick wall (in Gazebo) to make real environment.

**Command to copy brick_wall world environment.**
cp brick_wall.world empty.world

**Launching the scripts:**
Once the environment variables are set, you can run local training using the ros2 launch command.

1. For Local Training,
ros2 launch object_tracker_simulation local_training.launch.py

2. For Evalution,
ros2 launch object_tracker_simulation evalution.launch.py

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Rohita83 committed:
brick wall enviroment added and Noise filter is added in inference

timrobotson commented:
Can you make launching the brick wall environment a parameter to the launch file so that users don't need to copy it and overwrite empty_world. 

Could you also making adding noise a parameter so that users can easily turn it on or off.

Rohita83 committed:
set WORLD_NAME to empty or brick_wall to chooes enviroment

Rohita83 committed:
Patch2 : set WORLD_NAME to empty or brick_wall to chooes enviroment

Rohita83 commented:
> Can you make launching the brick wall environment a parameter to the launch file so that users don't need to copy it and overwrite empty_world.
> 
> Could you also making adding noise a parameter so that users can easily turn it on or off.

As per suggestion, I have made changes. 
 set environment variable WORLD_NAME to empty or brick_wall to choose environment

mm318 commented on robot_ws/src/turtlebot_controller/robomaker/inference_worker.py:
```diff
@@ -94,12 +95,16 @@ def load_graph(self):
         image = image.resize(TRAINING_IMAGE_SIZE)
         image = np.array(image)
 
+        if world_name=='brick_wall':
```
I think the logic here should be to do Gaussian blur based on a launch parameter AND when the world is not `empty`. The default value for the "`do_blur`" launch parameter can be `True`.

Rohita83 replied:
As per your suggestion, I have added DO_BLUR variable to control filter enable/disable 
f61daf1 . Default value set to true.  Also, we have added a model trained in a brick wall environment for reference b307578.

Rohita83 committed:
DO_BLUR enviroment variable added to enable/disable filter

Rohita83 committed:
Adding Model trainered in brick wall enviroment

mm318 commented:
When trying to run with `WORLD_NAME=brick_wall`, I'm getting an error like:
```
[robot_state_publisher-4] Could not open file [/home/robomaker/workspace/applications/simulation-application/dependencies/opt/ros/dashing/share/turtlebot3_description/urdf/turtlebot3_waffle_pi_brick_wall.urdf] for parsing.
```

Commit:
Fix for Error. related urdf file (brick wall urdf file not found)

Commit:
missed to delete unused variable

Commit:
Added WORLD_NAME environment variable in roboMakerSettings.json

Rohita83 commented:
> When trying to run with `WORLD_NAME=brick_wall`, I'm getting an error like:
> 
> ```
> [robot_state_publisher-4] Could not open file [/home/robomaker/workspace/applications/simulation-application/dependencies/opt/ros/dashing/share/turtlebot3_description/urdf/turtlebot3_waffle_pi_brick_wall.urdf] for parsing.
> ```

We found some issue with simulation_ws and modified accordingly. We have verified training/evaluation with brick_wall in localhost as well as robomaker. 
We have also added WORLD_NAME environment variable in roboMakerSettings.json

Could you please run it once robot_ws as well as simulation_ws (robomaker) at your end.

mm318 commented on robot_ws/src/turtlebot_controller/robomaker/inference_worker.py:
```diff
@@ -15,13 +15,14 @@
 import numpy as np
 import tensorflow as tf
-
+import cv2
```
Need to add `python3-opencv` as a dependency to the `package.xml`.

mm318 commented:
I have tested robot_ws and simulation_ws, both with and without `export WORLD_NAME=brick_wall`. Everything looks to be generally working. Please just fix the issue with `package.xml`. Thanks!

Commit:
Added python3-opencv in package.xml as suggested

Rohita83 commented:
> I have tested robot_ws and simulation_ws, both with and without `export WORLD_NAME=brick_wall`. Everything looks to be generally working. Please just fix the issue with `package.xml`.

Added python3-opencv in package.xml as suggested...
If everything is Ok. Could you please merge the PR.
